flink-connectors/flink-connector-base/src/main/java/org/apache/flink/connector/base/sink/AsyncSinkBaseBuilder.java

```diff
@@ -0,0 +1,115 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.flink.connector.base.sink;
+
+import org.apache.flink.annotation.PublicEvolving;
+import org.apache.flink.connector.base.sink.writer.ElementConverter;
+
+import java.io.Serializable;
+
+/**
+ * Abstract builder for constructing a concrete implementation of {@link AsyncSinkBase}.
+ *
+ * @param <InputT> type of elements that should be persisted in the destination
+ * @param <RequestEntryT> type of payload that contains the element and additional metadata that is
+ *     required to submit a single element to the destination
+ * @param <ConcreteBuilderT> type of concrete implementation of this builder class
+ */
+@PublicEvolving
+public abstract class AsyncSinkBaseBuilder<
+        InputT,
+        RequestEntryT extends Serializable,
+        ConcreteBuilderT extends AsyncSinkBaseBuilder<?, ?, ?>> {
+
+    protected ElementConverter<InputT, RequestEntryT> elementConverter;
+    protected Integer maxBatchSize;
+    protected Integer maxInFlightRequests;
+    protected Integer maxBufferedRequests;
+    protected Long flushOnBufferSizeInBytes;
+    protected Long maxTimeInBufferMS;
```

Comment:
Why are these protected rather than private?

Reply:
Wouldn't the concrete builder need to verify & use these like here `flink-connectors/flink-connector-aws/src/main/java/org/apache/flink/connector/kinesis/sink/KinesisDataStreamsSinkBuilder.java`

Reply:
Same as above